### ISSUE TITLE
Update functionality of Tree decoders

### ISSUE THREAD

**Issue by NoahAmsel** [clay-lab/transductions#20]
see the comments in models.py

**Comment by NoahAmsel** [comment]
Try multiplicative attention,
Make sure that my tree thing works when arity is set to 4
Try the hallucinated "inner" vocab if you want

**Comment by NoahAmsel** [comment]
Try my arithmetic task on the sequential decoder too

**Comment by NoahAmsel** [comment]
How to do back prop on hallucinated "inner" vocab words
take a softmax and take average of the embeddings associated with all the inner node vocab words weighted by that softmax. Then it's easy to backprop
